### PR TITLE
add json_path_match/2

### PR DESCRIPTION
**Pull request by lkarthee**
```elixir
s = Explorer.Series.from_list([~s({"a": 1}), ~s({"b": 1})])
#Explorer.Series<
  Polars[2]
  string ["{\"a\": 1}", "{\"b\": 1}"]
>
iex(3)> Explorer.Series.json_path_match(s, "$.a")
#Explorer.Series<
  Polars[2]
  string ["1", nil]
>
iex(4)> Explorer.Series.json_path_match(s, "$.")
** (RuntimeError) Polars Error: ComputeError(ErrString("error compiling JSONpath expression path error: \nEof\n"))
    (explorer 0.9.0-dev) lib/explorer/polars_backend/shared.ex:26: Explorer.PolarsBackend.Shared.apply_series/3
    iex:4: (file)
```

### CODE REVIEW

File: lib/explorer/backend/lazy_series.ex [hunk 1]
```diff
@@ -127,6 +127,7 @@ defmodule Explorer.Backend.LazySeries do
     substring: 3,
     split: 2,
     json_decode: 2,
```
Comment (lkarthee) [inline lib/explorer/backend/lazy_series.ex:129]: will add tests after review...

**Review by josevalim — APPROVED** [review]
LGTM!

**Comment by lkarthee** [comment]
added docs and tests

File: lib/explorer/series.ex [hunk 2]
```diff
@@ -6118,6 +6118,36 @@ defmodule Explorer.Series do
     apply_series(series, :json_decode, [dtype])
   end
 
+  @doc """
+  Extracts a string series using a [`JSONPath`](https://goessner.net/articles/JsonPath/) from a series.
```
Comment (josevalim) [inline lib/explorer/series.ex:6122]: Let's rename this to `json_path` and use it throughout the docs:

```suggestion
  Extracts a string series using a [`json_path`](https://goessner.net/articles/JsonPath/) from a series. 
```

File: lib/explorer/series.ex [hunk 3]
```diff
@@ -6118,6 +6118,36 @@ defmodule Explorer.Series do
...
+        Polars[3]
+        string [nil, nil, nil]
+      >
+
+  It raises an exception if the `JSONPath` is invalid.
+  """
+  @doc type: :string_wise
+  @spec json_path_match(Series.t(), String.t()) :: Series.t()
+  def json_path_match(%Series{dtype: :string} = series, path) do
+    apply_series(series, :json_path_match, [path])
```
Comment (josevalim) [inline lib/explorer/series.ex:6148]: And also in the signature:

```suggestion
  @spec json_path_match(Series.t(), String.t()) :: Series.t()
  def json_path_match(%Series{dtype: :string} = series, json_path) do
    apply_series(series, :json_path_match, [json_path])
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**